doc/elements-tx-format.md

```diff
@@ -0,0 +1,570 @@
+| Witness | Only if flags is 1 | Varies | `Witness` | | See [BIP 141](https://github.com/bitcoin/bips/blob/master/bip-0141.mediawiki). Note that Elements witnesses contain more data than Bitcoin witnesses. This extra data is described further below. |
+
+Serialized format: `[Version][Flags][Num Inputs][Inputs][Num Outputs][Outputs][Locktime][Witness]`
+
+This is the overarching structure of a serialized transaction. The rest of this document contains further details on specific parts, as well as examples.
+
+#### TxIn
+
+|Field|Required|Size|Data Type|Encoding|Notes| 
+|-----|--------|-----|---------|--------|-----|
+| Output Hash (TXID) | Yes | 32 bytes | `uint256` | Little-endian | Bytes of the output hash are encoded in little-endian. **NB**: this means that they appear backwards when serialized. |
+| Output Index | Yes | 4 bytes | `uint32_t` | Little-endian | **Input is a coinbase**: `0xffffffff`<br><br>The two most significant bits are reserved for flags.<br><br>**Input is a pegin:** second most significant bit is 1.<br><br>**Input has an asset issuance:** most significant bit is 1. |
+| ScriptSig Length | Yes | Varies | `VarInt` | | Set to `0x00` if the transaction is SegWit and the witness contains the signature. |
+| ScriptSig | If ScriptSig Length is non-zero | Varies | `hex` | Big-endian | |
+| Sequence | Yes | 4 bytes | `uint32_t` | Little-endian | |
+| Asset Issuance | Only if the transaction has an issuance (as indicated by the Output Index) | Varies | `AssetIssuance` |  | |
+
+Serialized format: `[Output Hash][Output Index][ScriptSig Length][ScriptSig][Sequence][Asset Issuance]`
+
+#### TxOut
+
+|Field|Required|Size|Data Type|Encoding|Notes| 
+|-----|--------|-----|---------|--------|-----|
+| Asset | Yes | 33 bytes | `ConfidentialAsset` | | Cannot be null. |
+| Amount | Yes | 9 or 33 bytes | `ConfidentialAmount` | | Cannot be null. |
+| Nonce | Yes | 1 or 33 bytes | `ConfidentialNonce` | | |
+| ScriptPubKey Length | Yes | Varies | `VarInt` | | | |
+| ScriptPubKey | If ScriptPubKey Length is non-zero | Varies | `hex` | Big-endian | |
+
+Serialized format: `[Asset][Amount][Nonce][ScriptPubKey Length][ScriptPubKey]`
+
+#### Witness
+
+As noted above, the witness is only present for SegWit transactions.
+
+|Field|Required|Size|Data Type|Encoding|Notes| 
+|-----|--------|-----|---------|--------|-----|
+| Input Witnesses| Yes | Varies | `Vector<InputWitness>` | | There is exactly one input witness for each input in the transaction.<br><br>This number is not explicitly included in the witness — it is implied by the number of inputs. |
+| Output Witnesses| Yes | Varies | `Vector<OutputWitness>` | | There is exactly one output witness for each output in the transaction.<br><br>This number is not explicitly included in the witness — it is implied by the number of outputs. |
+
+Serialized format: `[Input Witnesses][Output Witnesses]`
+
+#### InputWitness
+
```

Comment:
Might be helpful to add a one sentence description of this field, in case someone is arriving here without the previous context.